src/core_plugins/elasticsearch/lib/check_es_version.js

```diff
@@ -50,16 +52,19 @@ module.exports = function checkEsVersion(server, kibanaVersion) {
         ip: node.ip,
       }));
 
-      server.log(['warning'], {
-        tmpl: (
-          'You\'re running Kibana <%= kibanaVersion %> with some newer versions of ' +
-          'Elasticsearch. Update Kibana to the latest version to prevent compatibility issues: ' +
-          '<%= getHumanizedNodeNames(nodes).join(", ") %>'
-        ),
-        kibanaVersion,
-        getHumanizedNodeNames,
-        nodes: simplifiedNodes,
-      });
+      const warningNodeNames = getHumanizedNodeNames(simplifiedNodes).join(', ');
+      if (lastWarnedAboutNodes.get(server) !== warningNodeNames) {
```

Comment:
Is server a reference to the same object every time this function is called? If so maybe a `weakMap` is overkill, and we just cache a reference with a regular variable, e.g. `prevWarningNodeNames !== warningNodeNames`.